lib/oauth-db.js

```diff
@@ -38,65 +58,117 @@ OAuthDB.prototype.generateAuthCode = function(clientId, userId, scopes, expiresA
 
     var authCode = random.toString('hex');
 
-    this.authCodes[authCode] = {
-      client_id: clientId,
-      user_id: userId,
-      scopes: scopes,
-      expires_at: expiresAt
-    };
-
-    callback(null, authCode);
+    var saltedCode = crypto.createHash('sha256')
```

Comment:
I *think* we want to switch this to an [hmac generator](https://nodejs.org/api/crypto.html#crypto_crypto_createhmac_algorithm_key). Basically you input the salt as the key in the hmac instead. I want to find a concrete resource for *why* we should do this, but this comment on HN seems to explain the difference the best, and why we should use HMAC: https://news.ycombinator.com/item?id=1998247